tests/main/hidden-snap-dir/task.yaml

```diff
@@ -0,0 +1,94 @@
+summary: Check that the experimental hidden dir feature migrates the dir
+
+environment:
+    BLOB_DIR: $(pwd)/fake-store-blobdir
+    SNAP_NAME: test-snapd-tools
+
+prepare: |
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+
+    snap install "$SNAP_NAME"
+
+    setup_fake_store "$BLOB_DIR"
+
+restore: |
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    teardown_fake_store "$BLOB_DIR"
+    rm -rf "$BLOB_DIR"
+
+    snap set system experimental.hidden-snap-folder!
+    snap remove "$SNAP_NAME"
```

Comment:
what reminds me the tests/lib/reset.sh needs to be updated to remove /root/.snap and /home/test/.snap